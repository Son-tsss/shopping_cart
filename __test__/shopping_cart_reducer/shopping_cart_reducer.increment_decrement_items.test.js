import shoppingCartReducer from "../../src/shopping_cart_reducer"

it('increments existing item', () => {
    const prevState = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 1,
                discount: 1
            }
        ],
        discount: 1
    }

    const real = shoppingCartReducer(prevState, {type: "INCREMENT_ITEM", id: "i2"});

    const expected = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 2,
                discount: 1
            }
        ],
        discount: 1
    }

    expect(real).toEqual(expected);
});

it('incrementing not existing item chages nothing', () => {
    const prevState = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 1,
                discount: 1
            }
        ],
        discount: 1
    }

    const real = shoppingCartReducer(prevState, {type: "INCREMENT_ITEM", id: "i3"});

    expect(real).toMatchSnapshot();
});

it('decrements existing item', () => {
    const prevState = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 2,
                discount: 1
            }
        ],
        discount: 1
    }

    const real = shoppingCartReducer(prevState, {type: "DECREMENT_ITEM", id: "i2"});

    expect(real).toMatchSnapshot();
});

it('doesn`t decrement existing item below 0', () => {
    const prevState = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 0,
                discount: 1
            }
        ],
        discount: 1
    }

    const real = shoppingCartReducer(prevState, {type: "DECREMENT_ITEM", id: "i2"});

    expect(real).toMatchSnapshot();
});

it('decrementing not existing item changes nothing', () => {
    const prevState = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 1,
                discount: 1
            }
        ],
        discount: 1
    }

    const real = shoppingCartReducer(prevState, {type: "DECREMENT_ITEM", id: "i3"});

    expect(real).toMatchSnapshot();
});