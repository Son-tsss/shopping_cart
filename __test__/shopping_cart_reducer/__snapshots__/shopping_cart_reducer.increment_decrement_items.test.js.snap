// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decrementing not existing item changes nothing 1`] = `
Object {
  "discount": 1,
  "items": Array [
    Object {
      "count": 2,
      "discount": 0.5,
      "id": "i1",
      "price": 100,
    },
    Object {
      "count": 1,
      "discount": 1,
      "id": "i2",
      "price": 200,
    },
  ],
}
`;

exports[`decrements existing item 1`] = `
Object {
  "discount": 1,
  "items": Array [
    Object {
      "count": 2,
      "discount": 0.5,
      "id": "i1",
      "price": 100,
    },
    Object {
      "count": 1,
      "discount": 1,
      "id": "i2",
      "price": 200,
    },
  ],
}
`;

exports[`doesn\`t decrement existing item below 0 1`] = `
Object {
  "discount": 1,
  "items": Array [
    Object {
      "count": 2,
      "discount": 0.5,
      "id": "i1",
      "price": 100,
    },
    Object {
      "count": 0,
      "discount": 1,
      "id": "i2",
      "price": 200,
    },
  ],
}
`;

exports[`incrementing not existing item chages nothing 1`] = `
Object {
  "discount": 1,
  "items": Array [
    Object {
      "count": 2,
      "discount": 0.5,
      "id": "i1",
      "price": 100,
    },
    Object {
      "count": 1,
      "discount": 1,
      "id": "i2",
      "price": 200,
    },
  ],
}
`;
